x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/use_rule_preview_flyout.tsx

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ReactNode } from 'react';
+import React, { useCallback, useState, useMemo } from 'react';
+import type { EuiTabbedContentTab } from '@elastic/eui';
+import { invariant } from '../../../../../common/utils/invariant';
+import type { RuleObjectId } from '../../../../../common/api/detection_engine';
+import type { RuleResponse } from '../../../../../common/api/detection_engine/model/rule_schema';
+import { RuleDetailsFlyout } from '../../../rule_management/components/rule_details/rule_details_flyout';
+import { PREBUILT_RULE_UPDATE_FLYOUT_ANCHOR } from './upgrade_prebuilt_rules_table/upgrade_prebuilt_rules_table_context';
+
+interface UseRulePreviewFlyoutParams {
+  rules: RuleResponse[];
+  ruleActionsFactory: (rule: RuleResponse, closeRulePreview: () => void) => ReactNode;
+  extraTabsFactory?: (rule: RuleResponse) => EuiTabbedContentTab[];
+}
+
+interface UseRulePreviewFlyoutResult {
+  rulePreviewFlyout: ReactNode;
+  openRulePreview: (ruleId: RuleObjectId) => void;
+  closeRulePreview: () => void;
+}
+
+export function useRulePreviewFlyout({
+  rules,
+  extraTabsFactory,
+  ruleActionsFactory,
+}: UseRulePreviewFlyoutParams): UseRulePreviewFlyoutResult {
+  const [rule, setRuleForPreview] = useState<RuleResponse | undefined>();
+  const closeRulePreview = useCallback(() => setRuleForPreview(undefined), []);
+  const ruleActions = useMemo(
+    () => rule && ruleActionsFactory(rule, closeRulePreview),
+    [rule, ruleActionsFactory, closeRulePreview]
+  );
+  const extraTabs = useMemo(
+    () => (rule && extraTabsFactory ? extraTabsFactory(rule) : []),
+    [rule, extraTabsFactory]
+  );
+
+  return {
+    rulePreviewFlyout: rule && (
+      <RuleDetailsFlyout
+        rule={rule}
+        size="l"
+        id={PREBUILT_RULE_UPDATE_FLYOUT_ANCHOR}
+        dataTestSubj="updatePrebuiltRulePreview"
```

Comment:
We need to allow passing `dataTestSubj` via props, because we used: 
 - `dataTestSubj="installPrebuiltRulePreview"` for installation
 - `dataTestSubj="updatePrebuiltRulePreview"` for upgrade

We have a [Cypress test](https://github.com/maximpn/kibana/blob/3425c1333b09885c5f1feae701517ccff56eaf8f/x-pack/test/security_solution_cypress/cypress/e2e/detection_response/rule_management/prebuilt_rules/prebuilt_rules_preview.cy.ts) that uses these, but it doesn't run because it has issues with tags. I have created a [PR](https://github.com/elastic/kibana/pull/191978) to make it run again. We'll need to run these changes through CI once that PR is merged.

Reply:
Also `id={PREBUILT_RULE_UPDATE_FLYOUT_ANCHOR}` is not needed for the installation use case. I'm thinking maybe we could have a generic prop like `flyoutProps` that we would spread over? wdyt?

Reply:
Yes, thanks for noticing. I planned to address data test subject ids in follow up PRs where we add tests. As you can see the current code doesn't cause any failures.

Adding `flyoutProps` right away looks as a good option.